src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/building/ToolBuildingRecipe.java

```diff
@@ -37,7 +37,7 @@
 
   @Override
   public boolean matches(ITinkerStationInventory inv, World worldIn) {
-    return ToolBuildHandler.canToolBeBuilt(inv.getAllInputStacks(), this.output);
+    return ToolBuildHandler.canToolBeBuilt(inv.getAllInputStacks(), this.output) && inv.getTinkerableStack().isEmpty();
```

Comment:
Check the tinkerable stack first, that will run a little faster than the "can tool be built" logic